### ISSUE TITLE
Consider adding an option to disable Ansible host key checking

### ISSUE THREAD

**Issue by anzoman** [xlab-si/xopera-opera#89]
## Description
Ansible has host key checking enabled by default which also applies when running Ansible playbooks as TOSCA actuators with opera. But sometimes this can bring some problems, for instance with use cases when a new virtual machine gets spinned up for every new deployment. This means that user may connect to the VM for the first time when he initiates the deployment so he may be prompted to type in `[yes/no]` to continue connecting to the VM. This can be an issue for example when opera deployment are running in CI/CD pipelines and simple if user doesn't want to type in anything and just wants to continue with the deployment. In this case it might be helpful if he could disable that prompting by setting Ansible host key checking to false. But again this could not be done globally because checking host keys protects users against server spoofing and MITM attacks. So in cases when user understands the possible risks, it would be better to have an environment variable to control that behaviour (like `OPERA_SSH_HOST_KEY_CHECKING`) which can be set to false.

## Steps
To add this feature we need to update opera's Ansible executor to modify `ansible.cfg` file. This can be done by adding similar code snippet as the one that is below:

```python
opera_ssh_host_key_checking = os.environ.get("OPERA_SSH_HOST_KEY_CHECKING")
if opera_ssh_host_key_checking is not None:
    check = str(opera_ssh_host_key_checking).lower().strip()
    if check[:1] == 'f' or check[:1] == "false":
        with open("{}/ansible.cfg".format(dir_path), "w") as fd:
            fd.write("[defaults]\n")
            fd.write("host_key_checking = False\n")
```

Another possible way would be to set env var `ANSIBLE_HOST_KEY_CHECKING` to False.

## Current behaviour
Right now opera doesn't have the ability to disable Ansible host key checking (opera currently just sets `StrictHostKeyChecking` to no, which means that opera will automatically add new host keys to the user known hosts files).

## Expected results
The orchestrator could allow disabling host key checking.

**Comment by duelle** [comment]
As we need to spin up instances without manual interaction, I'm very much in favor of this!

**Comment by anzoman** [comment]
We are exploring this issue. Currently, it seems that this will be the solution: https://docs.ansible.com/ansible/latest/user_guide/connection_details.html#managing-host-key-checking.

**Comment by sstanovnik** [comment]
Resolved in #114.
